native/cocos/event/EventBus.h

```diff
@@ -0,0 +1,1165 @@
+            }
+        }
+
+        return false;
+    }
+
+protected:
+    std::vector<std::unique_ptr<TargetEventListenerBase>> _bubblingHandlers;
+    std::vector<std::unique_ptr<TargetEventListenerBase>> _capturingHandlers;
+    target_event_id_type _handlerId{};
+};
+
+} // namespace ebus
+
+#define DECLARE_TARGET_EVENT0(EventType, EmitterType, EventValue)                                                                  \
+    class EventType final : public ebus::TargetEventBaseHandler<EmitterType, static_cast<int>(EventValue), decltype(EventValue)> { \
+    public:                                                                                                                        \
+        using base_type = ebus::TargetEventBaseHandler<EmitterType, static_cast<int>(EventValue), decltype(EventValue)>;           \
+        using event_type = ebus::Event<EventType>;                                                                                 \
+        using persist_function_type = std::function<void(EmitterType *, event_type *)>;                                            \
+        using handler_type = std::function<void(event_type *)>;                                                                    \
+        constexpr static const char *EVENT_NAME = #EventType;                                                                      \
+        const char *getEventType() const { return typeid(typename base_type::event_enum_type).name(); }                            \
+    }
+
+#define DECLARE_TARGET_EVENT(EventType, EmitterType, EventValue, ...)                                                                           \
+    class EventType final : public ebus::TargetEventBaseHandler<EmitterType, static_cast<int>(EventValue), decltype(EventValue), __VA_ARGS__> { \
+    public:                                                                                                                                     \
+        using base_type = ebus::TargetEventBaseHandler<EmitterType, static_cast<int>(EventValue), decltype(EventValue), __VA_ARGS__>;           \
+        using event_type = ebus::Event<EventType>;                                                                                              \
+        using persist_function_type = std::function<void(EmitterType *, event_type *)>;                                                         \
+        using handler_type = std::function<void(event_type *)>;                                                                                 \
+        constexpr static const char *EVENT_NAME = #EventType;                                                                                   \
+        const char *getEventType() const { return typeid(typename base_type::event_enum_type).name(); }                                         \
+    }
+
+#define DECLARE_TARGET_EVENT1(EventType, EmitterType, EventValue, arg_type_0) \
+    DECLARE_TARGET_EVENT(EventType, EmitterType, EventValue, arg_type_0)
```

Comment:
Could we just use DECLARE_TARGET_EVENT for arguments and DECLARE_TARGET_EVENT0 without arguments?